### ISSUE TITLE
Logout not functioning across multiple subdomains

### ISSUE THREAD

**Issue by nloding** [IdentityServer/IdentityServer4#2511]
### Issue / Steps to reproduce the problem

I have an application that has multiple subdomains - for example, portal.mysite.com, admin.mysite.com, etc. - and they all use auth.mysite.com, which is running IdentityServer 4. The main site is using React with oidc-client-js, and the whole thing is set up for implicit flow. When the site transitions between subdomains, I call `oidcUserManager.signinSilent()` before loading the rest of the app to ensure the user is logged in.

This all works great. Logging out, however, doesn't work. If I logout from www.mysite.com then go back to www.mysite.com, I'm logged out. But if I logout from www.mysite.com then open portal.mysite.com, the silent authentication succeeds. I am calling logout with:

```js
const id_token_hint = user.id_token
oidcUserManager.signoutRedirect({ id_token_hint })
```

This properly redirects me to the logout page, where I've added a ton of redundant calls:

```csharp
var vm = await _account.BuildLoggedOutViewModelAsync(model.LogoutId);
await _signInManager.SignOutAsync();
await HttpContext.SignOutAsync(IdentityConstants.ExternalScheme);
await HttpContext.SignOutAsync(IdentityServerConstants.DefaultCookieAuthenticationScheme);
await HttpContext.SignOutAsync();
```

My understanding is that, if the cookie that IdentityServer uses is destroyed, the user should be logged out across everything. As browser security properly works, I cannot clear the oidc keys from browser storage across all the possible subdomains they may have visited.

Any suggestions on how I might log the user out of all the subdomains?

ps - I have not included a log, but can if needed. There are no errors logged, no exceptions thrown, so it seemed unnecessary.

**Comment by nloding** [comment]
I've played with cookie authentication schemes, none made a difference. I've manually deleted all the cookies, same issue. I've also used the interaction service to manually revoke all tokens for the session, and I still get logged in just fine.

I admit, I don't fully understand the back channel/front channel stuff, but the issue I'm trying to solve here is all within one tab - the app isn't running in the browser at the time of log out. I suppose it could be, but that's for later! I think?
